cabal-install/src/Distribution/Client/Configure.hs

```diff
@@ -299,6 +299,7 @@ configureSetupScript
       , useDependenciesExclusive = not defaultSetupDeps && isJust explicitSetupDeps
       , useVersionMacros = not defaultSetupDeps && isJust explicitSetupDeps
       , isInteractive = False
+      , isMainLibOrExeComponent = False
```

Comment:
Probably

```suggestion
      , isMainLibComponent = False
```

judging from the pattern matching below

Reply:
I've inverted the Bool setting, thanks for pointing this out!